File: C++17/백준/Silver/2563. 색종이/색종이.cc
#include<bits/stdc++.h>
using namespace std;
int n;
int arr[104][104];
int main(){
	cin>>n;
	int a,b;
	while(n--){
		cin>>a>>b;
		int l = 10;
		for(int i=b;i<l+b;i++){
			for(int j=a;j<l+a;j++){
				arr[i][j] = 1;
			}
		}
	}
	
	int cnt =0;
	for(int i=0;i<100;i++){
		for(int j=0;j<100;j++){
			if(arr[i][j] ==1) cnt++;
//			cout<<arr[i][j]<<' ';
		}
		//cout<<'\n';
	}
	cout<<cnt;
}

File: Java/백준/Bronze/10163. 색종이/색종이.java
//package im.q10163;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] arr = new int[1004][1004];
        int[] cnt = new int[104];
        for (int i = 1; i <=n ; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int c = Integer.parseInt(st.nextToken());
            int d = Integer.parseInt(st.nextToken());
            for(int j=a;j<a+c;j++){
                for(int k=b;k<b+d;k++){
                    arr[j][k]=i;
                }
            }
        }
        for (int i = 0; i < 1001; i++) {
            for (int j = 0; j < 1001; j++) {
                cnt[arr[i][j]]++;
            }
        }
        for (int i = 1; i <=n; i++) {
            System.out.println(cnt[i]);

        }



    }
}